model/fieldtypes/StringField.php

```diff
@@ -83,7 +83,9 @@ public function getNullifyEmpty() {
 	 * @see core/model/fieldtypes/DBField#exists()
 	 */
 	public function exists() {
-		return ($this->value || $this->value == '0') || ( !$this->nullifyEmpty && $this->value === '');
+		return $this->value
+			|| (is_string($this->value) && strlen($this->value))
+			|| (!$this->nullifyEmpty && $this->value === '');
```

Comment:
Yeah I thought about that, but it wasn't used consistently.

I had a look through the code and it looks like it's ment to map '' to nulls internally when saved to the database, but it's not clear what other values are ment to be mapped to (bool)false, true, int, etc....

The main problem I have with the API in this area is that we are taking a database field and using it as a casting helper. In practice a class should be single purpose, and the whole concept of casting for a template should be separate to how it's stored in the database.

I would like to separate these at some point in the future, but for now let's just make them work the best they can without changing the API too drastically.